src/Testing/src/xunit/AspNetTestInvoker.cs

```diff
@@ -59,6 +68,43 @@ protected override async Task<decimal> InvokeTestMethodAsync(object testClassIns
             return time;
         }
 
+        protected async Task<decimal> RetryAsync(RetryAttribute retryAttribute, object testClassInstance)
+        {
+            var attempts = 0;
+            var timeTaken = 0.0M;
+            for (attempts = 0; attempts < retryAttribute.MaxRetries; attempts++)
+            {
+                timeTaken = await base.InvokeTestMethodAsync(testClassInstance);
+                if (!Aggregator.HasExceptions)
+                {
+                    return timeTaken;
+                }
+                else if (attempts < retryAttribute.MaxRetries - 1)
+                {
+                    Aggregator.Clear();
```

Comment:
Any value in having a Task.Delay between retries?

Reply:
Sure we can add a delay, generally I think the helix machines aren't doing anything else other than running our tests, but I guess we do run our tests in parallel so maybe it wouldn't hurt to delay a bit, any opinions on how long to delay?  like 30 seconds?

Reply:
I was thinking of something of a couple of seconds.

Reply:
Ok, going with 5 second delay for now